common-healing.lic

```diff
@@ -124,47 +99,38 @@ module DRCH
     end
 
     pause 0.5
-    health_lines = reget(100).map(&:strip).reverse
-
-    # Example of what we're parsing:
-    # ===
-    # Corgar's injuries include...
-    # Wounds to the LEFT ARM:
-    # Fresh External:  cuts and bruises about the left arm -- more than minor
-    # Scars External:  severe scarring and chunks of flesh missing from the left arm -- very severe
-    # Fresh Internal:  minor swelling and bruising around the left arm -- more than minor
-    # Scars Internal:  a completely paralyzed left arm -- useless
-    # ===
-    if health_lines.grep(/^(?<name>.*)'s injuries include\b/).any?
-      target_name = Regexp.last_match[:name]
-      stop_line = "#{target_name}'s injuries include..."
-      perceived_wounds = parse_perceived_health(health_lines, stop_line)
-    end
-
-    # Now go through the perceived wounds map (key=body part)
-    # and bucket the wounds by severity into the 'wounds' map (key=severity).
-    wounds = Hash.new { |h, k| h[k] = [] }
-    perceived_wounds.values.flatten.each do |wound|
-      wounds[wound.severity] << wound
-    end
+    perceived_health_data = parse_perceived_health(target)
 
     score = calculate_score(wounds)
     waitrt? # This is probably not needed, touching others doesn't seem to cause RT in normal situations.
 
     return {
-      'wounds' => wounds,
+      'wounds' => perceived_health_data.wounds,
+      'poisoned' => perceived_health.poisoned,
       'score' => score
     }
   end
 
-  def parse_perceived_health(health_lines, stop_line)
+  # Examples of what this parses can be found in test/test_check_health.rb
+  def parse_perceived_health(target = nil)
+
+    stop_line = target.nil? ? 'Your injuries include...' : "#{target}'s injuries include..."
+    health_lines = reget(100).map(&:strip).reverse
+    if !health_lines.include?(stop_line)
+      print "exiting due to stop line: #{stop_line}"
```

Comment:
Is this a debug statement?

Reply:
Yes. I was having an issue with an extra reverse in my code so i put in some print lines.